File: src/main/java/com/vaadin/demo/component/checkbox/CheckboxGroup.java
package com.vaadin.demo.component.checkbox;

import com.vaadin.flow.component.checkbox.Checkbox;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-full-source-line

@Route("checkbox-group")
public class CheckboxGroup extends Div {

    public CheckboxGroup() {
        // tag::snippet[]
        Checkbox checkbox = new Checkbox();
        checkbox.setLabel("Enabled");

        add(checkbox);
        // end::snippet[]
    }

    public static class GridEditorExporter extends DemoExporter<CheckboxGroup> { // hidden-full-source-line
    } // hidden-full-source-line
}

File: frontend/demo/component/checkbox/checkbox-group.ts
import '../../init'; // hidden-full-source-line

import { html, LitElement, customElement } from 'lit-element';
import '@vaadin/vaadin-checkbox/vaadin-checkbox';
import '@vaadin/vaadin-checkbox/vaadin-checkbox-group';

@customElement('checkbox-group')
export class Example extends LitElement {
  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-checkbox-group label="Export data" theme="vertical">
        <vaadin-checkbox>Order ID</vaadin-checkbox>
        <vaadin-checkbox>Product name</vaadin-checkbox>
        <vaadin-checkbox>Customer</vaadin-checkbox>
        <vaadin-checkbox>Status</vaadin-checkbox>
      </vaadin-checkbox-group>
      <!-- end::snippet[] -->
    `;
  }
}
